packages/pigeon/lib/cpp_generator.dart

```diff
@@ -64,6 +65,31 @@ class CppOptions {
   }
 }
 
+/// Class that manages all Cpp header code generation.
+class CppHeaderGenerator extends Generator<CppOptions> {
```

Comment:
Don't we only want one generator for C++ and ObjC? We should set this up the way we want it now, because this is new public API and changing it later would be breaking (which isn't a huge deal for Pigeon, but there's no reason to set ourselves up to have to do it.

Dart outputs multiple files from the same generator, so presumably we can do the same here.